File: backend-go/internal/features/post/model/post.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Post struct {
	ID string `gorm:"primaryKey"`

	Title       string   `gorm:"size:50;not null;"`
	Description string   `gorm:"size:750; not null"`
	Tags        []string `gorm:"serializer:json" json:"tags"`

	CreatedAt time.Time
	UpdatedAt time.Time
	DeletedAt gorm.DeletedAt `gorm:"index"`

	AuthorID string
}

File: frontend/src/types/Post.ts
export interface Post {
  id: string;
  title: string;
  description: string;
  tags: string[];
  authorId?: string;
  author?: string;
}

export interface PostCreate {
  title: string;
  description: string;
  tags: string[];
}

export interface PostUpdate {
  title: string;
  description: string;
  tags: string[];
}
